scripts/config.py

```diff
@@ -503,25 +514,19 @@ def _get_compatible_GCC(self, nvcc_version):
              '11.3', '11.2', '11.1', '11',
              '10.4', '10.3', '10.2', '10.1', '10',
              '9.4', '9.3', '9.2', '9.1', '9',
-             '8.5', '8.4', '8.3', '8.2', '8.1', '8',
-             '7.5', '7.4', '7.3', '7.2', '7.1', '7',
-             '6.5', '6.4', '6.3', '6.2', '6.1', '6',
-             '5.5', '5.4', '5.3', '5.2', '5.1', '5']
-        if 8.0 <= nvcc_version < 9.0:
-            return v[v.index('5.3'):], True
-        elif 9.0 <= nvcc_version < 9.2:
-            return v[v.index('5.5'):], True
-        elif 9.2 <= nvcc_version < 10.1:
-            return v[v.index('7.5'):], True
-        elif 10.1 <= nvcc_version <= 10.2:
+             '8.5', '8.4', '8.3', '8.2', '8.1', '8']
+        if 10.1 <= nvcc_version <= 10.2:
```

Comment:
Maybe a variable that acts as a table for cuda-gcc compatibilities would be a good idea so the function always does the same, and only the variable changes over time.
For example:
```Python
# Does the array order need to be in descending order or ascending?
v = ['12.2', '12.1', '11.3', '11.2', '11.1', '11', '10.4', '10.3', '10.2', '10.1', '10', '9.4', '9.3', '9.2', '9.1', '9', '8.5', '8.4', '8.3', '8.2', '8.1', '8']
CUDA_GCC_COMPATIBILITY = {
    '10.1-10.2': v[v.index('8.5'):],
    '11.0-11.1': v[v.index('9.5'):],
    ...
}

# Function iterates over dictionary keys, split key by '-', check if provided nvcc version is between those versions, and, if so, returns the value associated with thay key. If not, move on to the next key, and if no key matches, return v, False

Reply:
For me, the current method looks more readable, perhaps out of habit.

Reply:
Maybe slightly more readable, but then function needs to be modified manually each time we support/remove support for a CUDA or GCC version. This way, only a variable is modified, which is a "better programming practice".

Reply:
Done